### ISSUE TITLE
Wayland window is undecorated - has no close button

### ISSUE THREAD

**Issue by mulle-nat** [glfw/glfw#2202]
It's curious, that there is apparently no bug report for this. I am building the current glfw (commit dd8a678a66f1967372e5a5e3deac41ebf65ee127) examples with `-DGLFW_BUILD_WAYLAND=ON -DGLFW_BUILD_X11=OFF`. The resultant window has no close button and looks very different to how it appears, if built for X11 (even when running under Wayland). `GLFW_DECORATED` set to true or false has no effect.

![image](https://user-images.githubusercontent.com/1381995/195618552-7c4aafae-662f-4491-8c08-c3d1d6026785.png)

```
GNOME Version 42.4
OS Name: Ubuntu 22.04.1 LTS
Windowing System: Wayland
Graphics: nvidia-driver-515
``` 

There are lots of other problems with Wayland, but this one is the most prominent.

**Comment by mulle-nat** [comment]
For some background, on why this is so, see the Discourse discussion [Non-standard window under Wayland and other questions ](https://discourse.glfw.org/t/non-standard-window-under-wayland-and-other-questions/2173).

**Comment by dougbinks** [comment]
This is related to issue #1639 and would be fixed by PR #1693

**Comment by elmindreda** [comment]
Support for libdecor has now been added with #2285 and was included in the latest stable release.
